src/migration/1689851339253-PromoteDateTime.ts

```diff
@@ -0,0 +1,19 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class PromoteDateTime1689851339253 implements MigrationInterface {
+  name = 'PromoteDateTime1689851339253';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`DROP INDEX "IDX_post_flags_promoteToPublic"`);
+    await queryRunner.query(
```

Comment:
@capJavert mentioned something about using HASH for text. Should we consider a different index?

Reply:
@capJavert Any suggestion for this from your side?
You probably have more knowledge from your recent research.

Reply:
It comes down to do we want to do `>=` and `<=` queries on that flags column but since we have to use `text` because jsonb does not support date we can leave the `HASH` because we would not be able to for example say `flags->promoteToPublic > '2023-07-07'`.

Not sure if we would have the need to query all the promoted posts to specific date on API side but if we do in that case we might be better off to changing the type to `number` and store timestamp. 

Then we could also move to `BTREE` to support `>=` and `<=` on index level.

Reply:
We can store isostrings as numbers as well, I read that was the alternative.

Reply:
@capJavert // @idoshamun We could choose UNIX time, but not sure it implied further complications down the line?
https://stackoverflow.com/a/29269257 (it's a number basically)

Reply:
Yeah I twas thinking of UNIX time yes and then use BTREE. If we needed it for frontend in any case I think we can always transform on graphorm level.

Reply:
BTREE it is then